tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -56,6 +56,8 @@ sealed trait NativeConfig {
    */
   def embedResources: Boolean
 
+  def nativeSourcePlugins: Seq[NativeSourcePlugin]
```

Comment:
Is the `NativeConfig` the right place to keep these?

Reply:
I think that could be the only way we could use to pass extensions without the usage of reflection. 
I can imagine the following scenario: 
We can define an sbt plugin, which could automatically populate some fields of nativeConfig,  

```
object ScalaNativeRustIntegrationPlugin extends AutoPlugin {
  override def requires: Plugins = plugins.ScalaNativePlugin
  
  object autoImport {
   val rustIntegrationConfig = taskKey[RustPLuginConfig](???) 
   val rustIntegrationPlugin = taskKey[RustIntegarionPlugin](???)
   
   case class RustPluginConfig() 
   class RustIntegarionPlugin(config: RustPluginConfig) extends NativeSourcePlugin  
 }
}
```

and then in users code 

```
val foo = project.in(file("foo")))
  .enablePlugins(ScalaNativeRustIntegrationPlugin)
  .settings(
    rustIntegrationConfig := ???,
    nativeConfig ~= _.withNativeSourceCompilerPlugins(rustIntegrationPlugin.value)
  )
```

Reply:
Yes, actually we can make it even easier for the user :)

```scala
object ScalaNativeRustIntegrationPlugin extends AutoPlugin {
  override def requires: Plugins = plugins.ScalaNativePlugin
  
  object autoImport {
   val rustIntegrationConfig = settingKey[RustPluginConfig](???) 
   
   case class RustPluginConfig()
 }

  override def projectSettings = Seq(
    nativeConfig := {
      val plugin = new RustSourceCompilerPlugin(rustIntegrationConfig.value)
      nativeConfig.value.withNativeSourceCompilerPlugin(plugin)
    }
  )
}
```

Then user code is just:
```scala
val foo = project.in(file("foo"))
  .enablePlugins(ScalaNativeRustIntegrationPlugin)
  .settings(
    rustIntegrationConfig := ???,
  )
```